Makefile

```diff
@@ -49,7 +49,7 @@ check_pydocstyle:
 	pydocstyle $(python-packages)
 
 check_doc:
-	tools/check_doc.py
+	tools/check_doc.py doc/language_reference/index.rst doc/development_guide/index.rst doc/user_guide/index.rst
```

Comment:
I did this change to detect cases where no code blocks are found and checked at all (like in previous months). Right now, the tools errors out when not at least a single code block has been found. As we have RST files without code blocks, those would be flagged as invalid if we check all RST files we find. How do we want to cope with that? I'd rather prefer to forget a file in the list instead of silently omitting all checks...

Reply:
/cc @jklmnn

Reply:
We could change the check so that the tool only fails if didn't find a code block in any document (instead of doing this check separately for each document, as it is done now). The unit tests should also prevent getting in the situation where no code block is checked. If we still see a possibility that the tool could not do the right thing, we should probably extend the tests.

Reply:
I second that we should allow documents without code blocks and run the tool on all files. I think the likelihood for forgetting to add a file is higher than a code block that is not detected by this tool going completely unnoticed (IMO this case mostly happens when there is a typo in the code block declaration which should be caught by other tools).

Reply:
Alright, I changed the tool to check all `*.rst` files below `doc/`. It will now error out when none of the files contained a code block. I added respective test cases to check that this mechanism actually works - this should be safe enough.